### PR TITLE
Mac_IL2CPP component installer

### LINKED ISSUE #125: Support Mac IL2CPP package install
IL2CPP is also supported on macOS standalone builds. Package path is `UnitySetup-Mac-IL2CPP-Support-for-Editor-$Version`.

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixes #125

If anyone knows where [the path for the Mac_IL2CPP playback engine](https://github.com/Microsoft/unitysetup.powershell/pull/161/files#diff-0eaeeb53ee5d7ee9f865be0f812aeaddR96) is we can address that before merge.
I don't have a mac to go and search it out on.

### CODE REVIEW

File: UnitySetup/UnitySetup.psm1 [hunk 1]
```diff
@@ -94,13 +93,15 @@ class UnitySetupInstance {
...
+                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "???");
                     [UnitySetupComponent]::Windows = , [io.path]::Combine("$playbackEnginePath", "WindowsStandaloneSupport");
-                    [UnitySetupComponent]::Linux = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
                 }
             }
         }
 
         # Common playback engines:
+        $componentTests[[UnitySetupComponent]::Linux] = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
+        $componentTests[[UnitySetupComponent]::Mac] = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport");
```
Comment (jwittner) [inline UnitySetup/UnitySetup.psm1]: I think it was more clear about what we were testing for before. With this change, on a mac, we'd set the component above, then we'd test this directory, which will never exist. Same as on Linux, when support comes. If we prefer performing tests that we know will always fail on some platforms then we should move the Windows test here as well.

File: UnitySetup/UnitySetup.psm1 [hunk 2]
```diff
@@ -318,7 +319,8 @@ function Find-UnitySetupInstaller {
         [UnitySetupComponent]::Facebook = , "$targetSupport/UnitySetup-Facebook-Games-Support-for-Editor-$Version.$installerExtension";
         [UnitySetupComponent]::Linux = , "$targetSupport/UnitySetup-Linux-Support-for-Editor-$Version.$installerExtension";
         [UnitySetupComponent]::Mac = "$targetSupport/UnitySetup-Mac-Support-for-Editor-$Version.$installerExtension",
-        "$targetSupport/UnitySetup-Mac-Mono-Support-for-Editor-$Version.$installerExtension";
+         "$targetSupport/UnitySetup-Mac-Mono-Support-for-Editor-$Version.$installerExtension";
```
Comment (jwittner) [inline UnitySetup/UnitySetup.psm1]: Are these the same component and they renamed it? Or can Mac-Mono-Support and Mac-Support exist side by side?

**Comment by jwittner** [comment]
We'll look into finding the Mac_IL2CPP path this week I hope.

File: UnitySetup/UnitySetup.psm1 (hunk 2 above)
Comment (StephenHodgson) [reply UnitySetup/UnitySetup.psm1]: nothing is renamed, not sure why that line changed, looks like just an extra space on the end. I can revert it, but it seemed to not have been formatted correctly or something.

Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
        "$targetSupport/UnitySetup-Mac-Mono-Support-for-Editor-$Version.$installerExtension";
```

File: UnitySetup/UnitySetup.psm1 (hunk 1 above)
Comment (StephenHodgson) [reply UnitySetup/UnitySetup.psm1]: Will they fail?

It looked like the exact same paths above, which is why I moved it down.

Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
```

File: UnitySetup/UnitySetup.psm1 [hunk 3]
```diff
@@ -79,8 +80,6 @@ class UnitySetupInstance {
                     [UnitySetupComponent]::UWP = [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_.NET_D3D"),
                     [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_D3D");
                     [UnitySetupComponent]::UWP_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_IL2CPP_D3D");
-                    [UnitySetupComponent]::Linux = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
-                    [UnitySetupComponent]::Mac = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport");
                 }
```
Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
                    [UnitySetupComponent]::Mac = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport");
                }
```

File: UnitySetup/UnitySetup.psm1 [hunk 4]
```diff
@@ -94,13 +94,14 @@ class UnitySetupInstance {
...
                     [UnitySetupComponent]::StandardAssets = , [io.path]::Combine("$Path", "Standard Assets");
+                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "???");
                     [UnitySetupComponent]::Windows = , [io.path]::Combine("$playbackEnginePath", "WindowsStandaloneSupport");
-                    [UnitySetupComponent]::Linux = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
                 }
             }
         }
 
         # Common playback engines:
+        $componentTests[[UnitySetupComponent]::Linux] = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
```
Comment (jwittner) [inline UnitySetup/UnitySetup.psm1]: Just to be nice to the future Linux implementers let's move this back into the Windows/Mac sections too.

File: UnitySetup/UnitySetup.psm1 (hunk 1 above)
Comment (jwittner) [reply UnitySetup/UnitySetup.psm1]: Well, it wouldn't throw an error, but on the mac platform, I don't believe the MacStandaloneSupport folder ever exists, etc.

File: UnitySetup/UnitySetup.psm1 [hunk 5]
```diff
@@ -94,13 +94,14 @@ class UnitySetupInstance {
                 @{
                     [UnitySetupComponent]::Documentation = , [io.path]::Combine("$Path", "Documentation");
                     [UnitySetupComponent]::StandardAssets = , [io.path]::Combine("$Path", "Standard Assets");
+                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "???");
                     [UnitySetupComponent]::Windows = , [io.path]::Combine("$playbackEnginePath", "WindowsStandaloneSupport");
```
Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
                    [UnitySetupComponent]::Windows = , [io.path]::Combine("$playbackEnginePath", "WindowsStandaloneSupport");
                    [UnitySetupComponent]::Linux = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
```

File: UnitySetup/UnitySetup.psm1 [hunk 6]
```diff
@@ -79,7 +80,6 @@ class UnitySetupInstance {
                     [UnitySetupComponent]::UWP = [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_.NET_D3D"),
                     [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_D3D");
                     [UnitySetupComponent]::UWP_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_IL2CPP_D3D");
```
Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
                    [UnitySetupComponent]::UWP_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "MetroSupport\Templates\UWP_IL2CPP_D3D");
                    [UnitySetupComponent]::Linux = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport");
```

File: UnitySetup/UnitySetup.psm1 (hunk 4 above)
Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
```

**Comment by awolowiecki720** [comment]
il2cppPath
Here's the path to the il2cpp folder 
@jwittner @StephenHodgson

**Comment by jwittner** [comment]
Hmmm, odd that it'd be under the ios folder.

**Comment by StephenHodgson** [comment]
@awolowiecki720 you sure that's correct for the Mac_IL2CPP?

**Comment by jwittner** [comment]
Alex is running a more controlled test now. Should have results today still.

**Comment by awolowiecki720** [comment]
packageContents
il2cppBefore
il2cppAfter

Actually, looks like this is the install location. Right-click on the Editor.app and "Show Package Contents". IL2CPP installs to this path (before and after pics) @jwittner @StephenHodgson

**Comment by StephenHodgson** [comment]
I wonder what each variation folder's full name is.

File: UnitySetup/UnitySetup.psm1 [hunk 7]
```diff
@@ -94,6 +95,7 @@ class UnitySetupInstance {
                 @{
                     [UnitySetupComponent]::Documentation = , [io.path]::Combine("$Path", "Documentation");
                     [UnitySetupComponent]::StandardAssets = , [io.path]::Combine("$Path", "Standard Assets");
+                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "???");
```
Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "A single folder name?");
```

Comment (jwittner) [reply UnitySetup/UnitySetup.psm1]: We use "windowsstandalonesupport\Variations\win32_development_il2cpp" for Windows_IL2CPP just above this line.

**Comment by jwittner** [comment]
Oh, I see, lol. @awolowiecki720 - can you drop in the path to the development_il2cpp folder? The name is collapsed in the images.

**Comment by awolowiecki720** [comment]
Oops. Yea it's 
/Applications/Unity/Unity.app/Contents/PlaybackEngines/MacStandaloneSupport/Variations/macosx64_development_il2cpp
@jwittner @StephenHodgson

File: UnitySetup/UnitySetup.psm1 [hunk 8]
```diff
@@ -94,6 +95,7 @@ class UnitySetupInstance {
                 @{
                     [UnitySetupComponent]::Documentation = , [io.path]::Combine("$Path", "Documentation");
                     [UnitySetupComponent]::StandardAssets = , [io.path]::Combine("$Path", "Standard Assets");
+                    #[UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport/Variations/macosx64_development_il2cpp");
```
Comment (jwittner) [inline UnitySetup/UnitySetup.psm1]: You'll need to uncomment this. =)

Comment (StephenHodgson) [inline UnitySetup/UnitySetup.psm1]:
```suggestion
                    [UnitySetupComponent]::Mac_IL2CPP = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport/Variations/macosx64_development_il2cpp");
```

**Comment by awolowiecki720** [comment]
Can verify that Mac_IL2CPP installs successfully, but I noticed 2 merge conflicts in your code when testing @StephenHodgson  
@jwittner what are next steps?

**Comment by jwittner** [comment]
@StephenHodgson If you can resolve your merge conflicts I'll merge this in. =)

**Comment by jwittner** [comment]
Actually I resolved the merge conflict through the cool GitHub tools. =)

**Pull request merged**